addons/stock/report/report_deliveryslip.xml

```diff
@@ -8,12 +8,12 @@
 
                 <t t-set="address">
                     <div name="div_outgoing_address">
-                        <div t-if="o.move_ids_without_package and o.move_ids_without_package[0].partner_id">
+                        <div t-if="o.move_ids_without_package and o.move_ids_without_package[0].partner_id and o.picking_type_id.code != 'incoming'">
                             <span><strong>Delivery Address:</strong></span>
                             <div t-field="o.move_ids_without_package[0].partner_id"
                                 t-options='{"widget": "contact", "fields": ["address", "name", "phone"], "no_marker": True, "phone_icons": True}'/>
                         </div>
-                        <div t-if="o.picking_type_id.code != 'internal' and (not o.move_ids_without_package or not o.move_ids_without_package[0].partner_id) and o.picking_type_id.warehouse_id.partner_id">
+                        <div t-if="o.picking_type_id.code != 'internal' and (not o.move_ids_without_package or not o.move_ids_without_package[0].partner_id or o.picking_type_id.code == 'incoming') and o.picking_type_id.warehouse_id.partner_id">
```

Comment:
Not sure we want to force the warehouse address in case of a receipt
@ticodoo Hi! Could you check this ?

Reply:
To stay consistent with the current behavior it's probably best we don't force it in stable.

That being said, I don't understand the current logic behind when the "Warehouse Address" should show as the right side address of the delivery slip (i.e. it looks like it only shows if it's a non-internal operation type that has "Move Entire Packages" active and all of the moves are in a package so it's possibly useless in almost all cases). This is maybe a thing for @thomasdeleval to consider for either a later improvement or a code removal cleanup in master since it looks like the logic this is based on is extremely old code (i.e. the picking operations report written 7+ years ago).